Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Search/SearchWidget.cpp

```diff
@@ -175,10 +175,21 @@ namespace AzToolsFramework
                 SetTypeFilters(buildTypesFilterList());
             }
 
-            AZ::IO::Path projectPath{ AZ::Utils::GetProjectPath() };
-            AZ::IO::Path enginePath{ AZ::Utils::GetEnginePath() };
-            auto engineFilter = new EngineFilter();
-            engineFilter->SetEngineAndProject(enginePath, projectPath);
+            const AZ::IO::Path enginePath{ AZ::Utils::GetEnginePath() };
+            AZ::IO::Path engineAssets = enginePath;
+            engineAssets = engineAssets.Append("Assets/Engine").LexicallyNormal();
+            AZ::IO::Path setregAssets = enginePath;
+            setregAssets = setregAssets.Append("Registry").LexicallyNormal();
+
+            auto filterFn = [engineAssets, setregAssets](const AssetBrowserEntry* entry)
+            {
+                AZ::IO::Path absolutePath = entry->GetEntryType() == AssetBrowserEntry::AssetEntryType::Product
+                    ? entry->GetParent()->GetFullPath()
+                    : entry->GetFullPath();
+
+                return !(absolutePath.IsRelativeTo(engineAssets) || absolutePath.IsRelativeTo(setregAssets));
+            };
```

Comment:
Correct me if I'm wrong but are we not doing a lot of string create/copies here?
Would it be more efficient if AssetBrowserEntry could provide GetFullPathView that returned an AZ::IO::PathView that we could use for the IsRelativeTo() call?  @lumberyard-employee-dm may have some additional thoughts but if we are running this filter on every file and folder it could be expensive no?

Reply:
Yeah an `AZ::IO::PathView` has the IsRelativeTo function which requires no string copies.

An alternative as well is to use an `const AZ::IO::Path&` variable if the `GetFullPath` function return as a const reference.

Reply:
Update: It does not, it returns an AZStd::string [by value](https://github.com/o3de/o3de/blob/b4ca369207868ba025f73edd8de71769610b9b37/Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntry.cpp#L193-L204).

Also because the function constructs a new path on the fly, returning an `AZ::IO::PathView` would be result in a dangling reference, so adding `GetFullPathView()` function should NOT be done.

The best way to avoid heap allocations is to return an `AZ::IO::FixedMaxPath`.
However a string copy would still need to occur.

If the desire is to avoid string copies, then the AssetBrowserEntry class would need to be update to cache off the full path as a member instead of performing an on the fly call to ResolvePath.

Reply:
Thanks! For now I'm happy to get this fix checked in, but am wary of performance concerns after running into the AssetBrowser freezing the Editor for minutes lately when processing many changes.  Since it's not a simple as using a reference or path view then we should probably not delay this fix and we could even do some proper profiling to determine how much the filter is affecting Editor perf.